### ISSUE TITLE
[BUG]: new scheduler ignoring scheduling parameters

### ISSUE THREAD

**Issue by Chetmn** [ankidroid/Anki-Android#15767]
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

Just updated to the latest ankidroid. The scheduling is broken, it is ignoring the intervals entered in the deck options.

Same deck works perfectly in the desktop app, and was working perfectly in the previous version.

### Expected behaviour

Ankidroid should be following the intervals specified in deck options like it always has.

### Actual behaviour

Completely ignores deck intervals.

### Debug info

```text
Version 2.17.0,
```

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by david-allison** [comment]
Hi, what version are you on?

**Comment by Chetmn** [comment]
Version 2.17.0, the latest version from F-Droid

BTW I actually rolled back to 2.16.15 as the scheduling works.

**Comment by david-allison** [comment]
Could you provide an export of your collection and zip it up?

I can accept this privately if you don't want to share this publicly

**Comment by mikehardy** [comment]
Note we are on 2.17.3, and strangely it appears that f-droid has [noticed it exists](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/com.ichi2.anki.yml) 

But it hasn't built it for whatever reason.

Built 2.17.0 (which you have) https://f-droid.org/repo/com.ichi2.anki_21700300.log.gz
Built 2.17.1 https://f-droid.org/repo/com.ichi2.anki_21701300.log.gz
2.17.2 was not formally released, we had a failure in our automated release systems
2.17.3 is in their metadata and is available on play now but fdroid hasn't built it? https://f-droid.org/repo/com.ichi2.anki_21703300.log.gz (404, for me)

**Comment by Chetmn** [comment]
I can share the collection privately if you show me how but it's pretty huge - it's over 15000 cards...

**Comment by Chetmn** [comment]
2.17.0 is the latest I see in F Droid and 2.16.15 is the version showing up in the playstore for me.

If you guys have a link for 2.17.3 I'm happy to test it to see if the bug is still there.

**Comment by mikehardy** [comment]
https://github.com/ankidroid/Anki-Android/releases is where we have everything

**Comment by Chetmn** [comment]
Unfortunately the bug is still in 2.17.3...

**Comment by david-allison** [comment]
![Screenshot 2024-03-06 at 14 40 20](https://github.com/ankidroid/Anki-Android/assets/62114487/501cd5bd-3eee-4fbe-94d6-e27b6ca6d544)
![Screenshot 2024-03-06 at 14 39 48](https://github.com/ankidroid/Anki-Android/assets/62114487/5a87dc95-48a5-4803-a8b7-024d6e480151)

@Chetmn I've tried two cards in Anki Desktop 23.10 and AnkiDroid 2.17.5 and they're showing the same scheduling values

1. Could you show a screenshot of a problematic card
2. Could you copy/paste AnkiDroid's debug information, and Anki Desktop's version

**Comment by Chetmn** [comment]
Thanks DavidSorry I was on a business trip and just got back.I installed 2.17.1 (latest version on f-droid) on an old phone so I could do a direct comparison at exactly the same time, with exactly the same database freshly downloaded from ankiweb. Screenshots attached.The 2.16 gives exactly the same intervals as my desktop (latest qt6 version), as the screenshot shows 2.17.1 gives very different intervals.On 7/3/24, 12:42 am David Allison ***@***.***> wrote:

   
  Screenshot.2024-03-06.at.14.40.20.png (view on web) Screenshot.2024-03-06.at.14.39.48.png (view on web) 
  @Chetmn I've tried two cards in Anki Desktop 23.10 and AnkiDroid 2.17.5 and they're showing the same scheduling values 
   
   Could you show a screenshot of a problematic card 
   Could you copy/paste AnkiDroid's debug information, and Anki Desktop's version

**Comment by david-allison** [comment]
Hi @Chetmn, no rush!

Your screenshots weren't transferred to GitHub, could you come to the issue and provide them. 

Thanks!

https://github.com/ankidroid/Anki-Android/issues/15767

**Comment by Chetmn** [comment]
![2 17 1](https://github.com/ankidroid/Anki-Android/assets/162118012/de5eef4a-0bcb-4e6c-bd77-e3993c2f5422)
![2 16 - correct same as desktop](https://github.com/ankidroid/Anki-Android/assets/162118012/69489861-6626-45dc-9f7b-4942f276a47e)

**Comment by david-allison** [comment]
@Chetmn 

* Could you show a screenshot with differences between Anki Desktop and AnkiDroid?
* Could you post the version of Anki Desktop that you're using?

**Comment by Chetmn** [comment]
I've found the error.

I had some old data in custom scheduling (I tried FSRS a while back, didn't work for me). FSRS is turned off so Ankidroid 2.16 and desktop (23.12.1 FWIW) were correctly ignoring the custom scheduling data and using the standard max interval, ease etc.

Ankidroid 2.17 has it backwards - it's using the custom scheduling data to override the standard intervals despite FSRS being turned off. Once I deleted that scheduling data 2.17 works as expected.

**Comment by dae** [comment]
The FSRS toggle has no influence over whether custom scheduling is applied. If it's in the deck options, it's always run - even on desktop.

**Comment by Chetmn** [comment]
All I know is that neither desktop, web, nor droid 2.16 ever gave me an interval greater than the maximim interval I had specified, and when I encountered the issue and checked they gave exactly the same intervals.

Droid 2.17 immediately, and consistently, gave me intervals both greater than the maximum interval I had set, and completely different to what the desktop, web and 2.16 were giving me.

As soon as I deleted that data droid 2.17 started giving me the same intervals as the desktop client and droid 2.16.

I'm not a programmer. All I can report is that was the single variable that solved the inconsistency problem.
